doculog/main.py

```diff
@@ -1,21 +1,94 @@
 """
 CLI entrypoints to the code.
 """
+from asyncio.log import logger
 from pathlib import Path
+from argparse import ArgumentParser
+import os
+import logging
+import sys
 
-from doculog.changelog import ChangelogDoc
 from doculog.config import configure
+from doculog import ChangelogDoc, __version__
 
 
-def generate_changelog():
+def generate_changelog(overwrite: bool = False):
     root = Path.cwd()
 
     config = configure(root)
     log_path = root / config["changelog_name"]
 
-    doc = ChangelogDoc(log_path)
+    if overwrite:
+        if log_path.exists():
+            os.remove(log_path.absolute())
+            logger.info("Overwriting current changelog")
+        else:
+            logger.warning("Skipping overwrite, existing changelog file not found", exc_info=1)
+
+    doc = ChangelogDoc(log_path, config["categories"], config["category_options"])
```

Comment:
Sorry, what are `categories` and `category_options` here? Is that from a different PR?

Reply:
woops. they are